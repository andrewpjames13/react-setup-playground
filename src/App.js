import React, { Component } from 'react';
import logo from './logo.svg';
import styles from './App.scss';
import grid from './grid.scss';

class App extends Component {
  render() {
    console.log(styles, grid);
    return (
      <div className={`${styles.App} ${grid.colXs12}`}>
        <div className={styles.header}>
          <img src={logo} className={styles.logo} alt="logo" />
          <h2>Welcome to React</h2>
        </div>
        <p className={styles.intro}>
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}
export default App;
