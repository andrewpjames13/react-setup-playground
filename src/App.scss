@import './index';
.App {
  text-align: center;
}

.logo {
  animation: logo-spin infinite 20s linear;
  height: 80px;
}

.header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.intro {
  font-size: large;
  &:hover {
    color: $green;
  }
}

@keyframes logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
