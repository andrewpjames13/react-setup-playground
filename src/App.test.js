import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import App from './App';

it('renders without crashing', () => {
  const wrapper = shallow(<App />);
  const welcome = <h2>Welcome to React</h2>;
  expect(wrapper.find('.App').hasClass('colXs12')).toEqual(true);
  expect(wrapper.contains(welcome)).toEqual(true);
});

test('snapshot', () => {
  const tree = renderer.create(
    <App/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
