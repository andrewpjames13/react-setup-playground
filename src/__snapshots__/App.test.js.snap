exports[`test snapshot 1`] = `
<div
  className="App colXs12">
  <div
    className="header">
    <img
      alt="logo"
      className="logo"
      src="test-file-stub" />
    <h2>
      Welcome to React
    </h2>
  </div>
  <p
    className="intro">
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;
